raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -67,6 +75,39 @@ protected LeaderState(
         }
         this.grantingVoters.addAll(grantingVoters);
         this.log = logContext.logger(LeaderState.class);
+        this.accumulator = Objects.requireNonNull(accumulator, "accumulator must be non-null");
+    }
+
+    public BatchAccumulator<T> accumulator() {
+        return this.accumulator;
+    }
+
+    private static List<Voter> convertToVoters(Set<Integer> voterIds) {
+        return voterIds.stream()
+            .map(follower -> new Voter().setVoterId(follower))
+            .collect(Collectors.toList());
+    }
+
+    public void appendLeaderChangeMessage(int epoch, long baseOffset, long currentTimeMs) {
+        List<Voter> voters = convertToVoters(this.followers());
```

Comment:
If you look at `followers`, it is using `voterStates.keySet` and then stripping out the leaderId. Then we add it back on line 96 below. It seems simpler to use `voterStates.keySet` directly.